transaction-pool/src/lib.rs

```diff
@@ -281,28 +151,32 @@ impl<'a, A> Verifier<'a, A> where
 	}
 }
 
-impl<'a, A> txpool::Verifier<UncheckedExtrinsic> for Verifier<'a, A> where
-	A: 'a + PolkadotApi,
+impl<A> extrinsic_pool::ChainApi for ChainApi<A> where
+	A: PolkadotApi + Send + Sync,
 {
-	type VerifiedTransaction = VerifiedTransaction;
+	type Block = Block;
+	type Hash = Hash;
+	type Sender = Option<AccountId>;
+	type VEx = VerifiedTransaction;
+	type Ready = HashMap<AccountId, u32>;
 	type Error = Error;
+	type Score = u64;
+	type Event = ();
 
-	fn verify_transaction(&self, uxt: UncheckedExtrinsic) -> Result<Self::VerifiedTransaction> {
+	fn verify_transaction(&self, at: &BlockId, xt: &ExtrinsicFor<Self>) -> Result<Self::VEx> {
+		let encoded = xt.encode();
+		let uxt = UncheckedExtrinsic::decode(&mut encoded.as_slice()).ok_or_else(|| ErrorKind::InvalidExtrinsicFormat)?;
 		if !uxt.is_signed() {
 			bail!(ErrorKind::IsInherent(uxt))
 		}
 
-		let encoded = uxt.encode();
-		let encoded_size = encoded.len();
-
+		let (encoded_size, hash) = (encoded.len(), BlakeTwo256::hash(&encoded));
```

Comment:
Previously we were avoiding hashing if it was too large, but I guess it's not an issue.